stage0/src/main.rs

```diff
@@ -284,6 +294,32 @@ pub extern "C" fn rust64_start(encrypted: u64) -> ! {
     }
     /* TODO(#3198): interrogate qemu for memory areas and set up the zero page. */
 
+    // Safety: We assume the VMM has filled in the basic zero page for us. This is not true with
+    // qemu. In the future, construction of the full zero page should be in here, and instead of
+    // `assume_init_mut` we will use the safe `write` to zero it out and get a mutable reference.
+    let zero_page = unsafe { BOOT_ZERO_PAGE.assume_init_mut() };
+
+    if snp {
```

Comment:
Why is this if-block not merged with the next? (I like the structure that it gives to the code, but I wonder if there is any other reason for it.)

Reply:
No other reason but that they're semantically different operations - this deals with setting up the zero page, the other block deals with `PVALIDATE`ing the memory.

Reply:
In particular, I expect the second block to go away in its current form in not too distant future.